### PR TITLE
Make docs.ms link relative link

### PR DESCRIPTION
**Pull request by sima-zhu**
https://raw.githubusercontent.com/Azure/azure-docs-sdk-java/main/docs-ref-services/preview/sdk-template-readme.md

Metadata definition: https://review.docs.microsoft.com/en-us/help/contribute/metadata-attributes?branch=main

### CODE REVIEW

File: eng/common/scripts/Update-DocsMsMetadata.ps1 [hunk 1]
```diff
@@ -129,13 +129,12 @@ author: $author
...
 ms.date: $date
 ms.topic: reference
-ms.prod: azure
-ms.technology: azure
 ms.devlang: $Language
 ms.service: $service
 ---
 "@
 
+  $ReadmeContent = $ReadmeContent -replace "https://docs.microsoft.com(/en-us)?", ""
```
Comment (danieljurek) [inline eng/common/scripts/Update-DocsMsMetadata.ps1:137]: This would erase links whose value is `https://docs.microsoft.com` or  `https://docs.microsoft.com/en-us` ... 

Running this in the console gives: 

```pwsh
 @"
https://docs.microsoft.com/
https://docs.microsoft.com

https://docs.microsoft.com/en-us/
https://docs.microsoft.com/en-us

https://docs.microsoft.com/test/
https://docs.microsoft.com/test

https://docs.microsoft.com/en-us/test/
https://docs.microsoft.com/en-us/test
"@ -replace "https://docs.microsoft.com(/en-us)?", ""
/
    # <-- there should be a / here

/
    # <-- there should be a / here

/test/
/test

/test/
/test
```

Another approach might be to include the trailing forward slash: 

```suggestion 
  $ReadmeContent = $ReadmeContent -replace "https://docs.microsoft.com(/en-us)?/?", "/"
```

Running in the console gives: 

```pwsh
@"
https://docs.microsoft.com/
https://docs.microsoft.com

https://docs.microsoft.com/en-us/
https://docs.microsoft.com/en-us

https://docs.microsoft.com/test/
https://docs.microsoft.com/test

https://docs.microsoft.com/en-us/test/
https://docs.microsoft.com/en-us/test
"@ -replace "https://docs.microsoft.com(/en-us)?/?", "/"
/
/

/
/

/test/
/test

/test/
/test
```

**Comment by sima-zhu** [comment]
> Can we get an example of this running and updating links? I think this can be accomplished by [setting `SetDevVersion` to `true`](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/stages/archetype-java-release.yml#L410) and queuing a build for a service pipeline.

I have pasted the test in direction.

**Review by danieljurek — APPROVED** [review]
LGTM!

I think one small corner case still exists here: we a language region-language code is used here that isn't `en-us` ... We could try to match against a [list](https://support.microsoft.com/en-us/topic/country-region-and-language-codes-add36afe-804a-44f1-ae68-cfb9c9b72f8b) of possible codes but I don't think the effort would be worth it. 

Searching through the Java codebase reveals that other language codes are not in use in our README.md files.

Furthermore those errors would be flagged as warnings in OPS and could be addressed on a case by case basis if we start seeing more pop up.

**Comment by weshaggard** [comment]
I thought about the other languages as well but based on our link guidance folks shouldn't even have the `/en-us` in the link either so I think we are probably ok here and we probably can even remove that from the regex if we wanted but I'm fine keeping as well.

File: eng/common/scripts/Update-DocsMsMetadata.ps1 [hunk 2]
```diff
@@ -129,13 +129,12 @@ author: $author
 ms.author: $msauthor
 ms.date: $date
 ms.topic: reference
```
Comment (weshaggard) [inline eng/common/scripts/Update-DocsMsMetadata.ps1:131]: Why were these removed?

**Review by weshaggard — APPROVED** [review]

File: eng/common/scripts/Update-DocsMsMetadata.ps1 (hunk 2 above)
Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsMetadata.ps1:131]: This is according the their metadata guidance. If we use the ms.service, then the set of `ms.prod` and `ms.technology` are not necessary. 
https://raw.githubusercontent.com/Azure/azure-docs-sdk-java/main/docs-ref-services/preview/sdk-template-readme.md

Comment (sima-zhu) [reply eng/common/scripts/Update-DocsMsMetadata.ps1:131]: There are many OPS warning regarding we are using wrong ms.prod value

**Comment by sima-zhu** [comment]
I think we should leave the other locale as it is, so people can redirect to the right language as they defined.
